test/features/test_gasd_estimation.cpp

```diff
@@ -0,0 +1,335 @@
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder(s) nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#include <gtest/gtest.h>
+#include <pcl/point_cloud.h>
+#include <pcl/features/gasd.h>
+#include <pcl/io/pcd_io.h>
+
+pcl::PointCloud<pcl::PointXYZ>::Ptr cloud;
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+void
+createColorCloud (pcl::PointCloud<pcl::PointXYZRGBA> &colorCloud)
+{
+  for (size_t i = 0; i < cloud->points.size (); ++i)
+  {
+    pcl::PointXYZRGBA p;
+    p.getVector3fMap () = cloud->points[i].getVector3fMap ();
+
+    p.rgba = ( (i % 255) << 16) + ( ( (255 - i) % 255) << 8) + ( (i * 37) % 255);
+    colorCloud.push_back (p);
+  }
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+TEST(PCL, GASDTransformEstimation)
+{
+  pcl::GASDEstimation<pcl::PointXYZ, pcl::GASDSignature512> gasd;
+  gasd.setInputCloud (cloud);
+
+  pcl::PointCloud<pcl::GASDSignature512> descriptor;
+  gasd.compute (descriptor);
+
+  Eigen::Matrix4f trans = gasd.getTransform ();
+
+  Eigen::Matrix4f ref_trans;
+  ref_trans << 0.661875, -0.704840, 0.255192, 0.0846344,
+  -0.748769, -0.605475, 0.269713, 0.0330151,
+  -0.035592, -0.369596, -0.928511, 0.0622551,
+  0, 0, 0, 1;
+
+  for (size_t i = 0; i < trans.rows(); ++i)
+  {
+    for (size_t j = 0; j < trans.cols (); ++j)
+    {
+      EXPECT_NEAR (trans (i, j), ref_trans (i, j), 1e-5);
+    }
+  }
+}
```

Comment:
What you did was correct but to suppress the warnings switch `i` and `j` types to `int`.